crates/bevy_scene/src/serde.rs

```diff
@@ -167,7 +170,18 @@ impl<'a> Serialize for SceneMapSerializer<'a> {
         S: Serializer,
     {
         let mut state = serializer.serialize_map(Some(self.entries.len()))?;
-        for reflect in self.entries {
+        let sorted_entries: Vec<&Box<dyn PartialReflect>> = self
+            .entries
+            .iter()
+            .sorted_by(|&reflect_a, &reflect_b| {
+                reflect_a
+                    .get_represented_type_info()
+                    .unwrap()
+                    .type_path()
+                    .cmp(reflect_b.get_represented_type_info().unwrap().type_path())
+            })
+            .collect();
+        for reflect in sorted_entries {
             state.serialize_entry(
                 reflect.get_represented_type_info().unwrap().type_path(),
                 &TypedReflectSerializer::new(reflect.as_partial_reflect(), self.registry),
```

Comment:
```suggestion
        let sorted_entries = {
            let mut entries = self
                .entries
                .iter()
                .map(|entry| {
                    (
                        entry.get_represented_type_info().unwrap().type_path(),
                        entry.as_partial_reflect(),
                    )
                })
                .collect::<Vec<_>>();
            entries.sort_by_key(|(type_path, _partial)| *type_path);
            entries
        };
        for (type_path, partial) in sorted_entries {
            state.serialize_entry(
                type_path,
                &TypedReflectSerializer::new(partial, self.registry),
```

Since this is being collected into a `Vec` anyway, why not omit `itertools` and just use [`Vec::sort_by_key`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.sort_by_key)? Also suggest this map to avoid a double-unwrap